docker_daemon/check.py

```diff
@@ -236,58 +237,69 @@ def init(self):
+                    except Exception as e:
+                        self.log.warning('Could not retrieve kubernetes labels: %s' % str(e))
 
-        # containers running with custom cgroups?
-        custom_cgroups = _is_affirmative(instance.get('custom_cgroups', False))
+            # containers running with custom cgroups?
+            custom_cgroups = _is_affirmative(instance.get('custom_cgroups', False))
 
-        # Get the list of containers and the index of their names
-        health_service_checks = True if self.whitelist_patterns else False
-        containers_by_id = self._get_and_count_containers(custom_cgroups, health_service_checks)
-        containers_by_id = self._crawl_container_pids(containers_by_id, custom_cgroups)
+            # Get the list of containers and the index of their names
+            health_service_checks = True if self.whitelist_patterns else False
+            containers_by_id = self._get_and_count_containers(custom_cgroups, health_service_checks)
+            containers_by_id = self._crawl_container_pids(containers_by_id, custom_cgroups)
 
-        # Send events from Docker API
-        if self.collect_events or self._service_discovery or not self._disable_net_metrics or self.collect_exit_codes:
-            self._process_events(containers_by_id)
+            # Send events from Docker API
+            if self.collect_events or self._service_discovery or not self._disable_net_metrics or self.collect_exit_codes:
+                self._process_events(containers_by_id)
 
-        # Report performance container metrics (cpu, mem, net, io)
-        self._report_performance_metrics(containers_by_id)
+            # Report performance container metrics (cpu, mem, net, io)
+            self._report_performance_metrics(containers_by_id)
 
-        if self.collect_container_size:
-            self._report_container_size(containers_by_id)
+            if self.collect_container_size:
+                self._report_container_size(containers_by_id)
 
-        if self.collect_container_count:
-            self._report_container_count(containers_by_id)
+            if self.collect_container_count:
+                self._report_container_count(containers_by_id)
 
-        if self.collect_volume_count:
-            self._report_volume_count()
+            if self.collect_volume_count:
+                self._report_volume_count()
 
-        # Collect disk stats from Docker info command
-        if self.collect_disk_stats:
-            self._report_disk_stats()
+            # Collect disk stats from Docker info command
+            if self.collect_disk_stats:
+                self._report_disk_stats()
 
-        if health_service_checks:
-            self._send_container_healthcheck_sc(containers_by_id)
+            if health_service_checks:
+                self._send_container_healthcheck_sc(containers_by_id)
+        except Exception as e:
+            self.log.critical(e)
```

Comment:
Please use [self.log.exception](https://docs.python.org/2/library/logging.html#logging.Logger.exception) instead, with the same message